src/components/AmplitudeAnalytics/Trace.tsx

```diff
@@ -0,0 +1,62 @@
+import { createContext, memo, PropsWithChildren, useContext, useEffect, useMemo } from 'react'
+
+import { sendAnalyticsEvent } from '.'
+import { ElementName, EventName, ModalName, PageName, SectionName } from './constants'
+
+export interface ITraceContext {
+  // Highest order context: eg Swap or Explore.
+  page?: PageName
+
+  // Enclosed section name. Can be as wide or narrow as necessary to
+  // provide context.
+  section?: SectionName | ModalName
+
+  // Element name mostly used to identify events sources
+  // Does not need to be unique given the higher order page and section.
+  element?: ElementName
+}
+
+export const TraceContext = createContext<ITraceContext>({})
+
+type TraceProps = {
+  shouldLogImpression?: boolean // whether to log impression on mount
+  eventName?: EventName
+  eventProperties?: Record<string, unknown>
+} & ITraceContext
+
+/**
+ * Sends an analytics event on mount (if shouldLogImpression is set),
+ * and propagates the context to child traces.
+ */
+export const Trace = memo(
+  ({
+    children,
+    shouldLogImpression,
```

Comment:
put `shouldLogImpression` above `eventName` so that they're ordered the same as they're defined (this makes it easier to follow for new readers)